.pre-commit-config.yaml

```diff
@@ -1,12 +1,61 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
 repos:
--   repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v2.3.0
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.5.0
     hooks:
-    -   id: check-yaml
-    -   id: end-of-file-fixer
-    -   id: trailing-whitespace
-    -   id: debug-statements
--   repo: https://github.com/psf/black
-    rev: 22.3.0
+      - id: check-symlinks
+      - id: destroyed-symlinks
+      - id: trailing-whitespace
+      - id: end-of-file-fixer
+      - id: check-yaml
+      - id: check-toml
+      - id: check-ast
+      - id: check-added-large-files
+      - id: check-merge-conflict
+      - id: check-executables-have-shebangs
+      - id: check-shebang-scripts-are-executable
+      - id: detect-private-key
+      - id: debug-statements
+#  - repo: https://github.com/codespell-project/codespell
+#    rev: v2.2.6
+#    exclude: ^(src/common)|(src/emucore)|(src/environment)|(src/games)
+#    hooks:
+#      - id: codespell
+#        args:
+#          - --ignore-words-list=
+  - repo: https://github.com/PyCQA/flake8
+    rev: 7.0.0
     hooks:
-    -   id: black
+      - id: flake8
```

Comment:
Should switch to ruff?

Reply:
We just copied in Gymansium's pre-commit, it certainly would be an option